code-challenges.test.js

```diff
@@ -42,13 +85,46 @@ const randomNouns1 = ["streetlamp", "potato", "teeth", "conclusion", "nephew"]
 // Expected output: ["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"]
 const randomNouns2 = ["temperature", "database", "chopsticks", "mango"]
 // Expected output: ["Temperature", "Database", "Chopsticks", "Mango"]
+// I will need to write the test function that takes in an array of words and returns an array with all the words capitalized.
+describe ("capAllLetters", () => {
+    it ("takes in an array of words and returns an array with all the words capitalized.", () => {
+      expect(capAllLetters(randomNouns1)).toEqual(["Streetlamp", "Potato", "Teeth", "Conclusion", "Nephew"])
+      expect(capAllLetters(randomNouns2)).toEqual(["Temperature", "Database", "Chopsticks", "Mango"])
+    })
+})
+//  ReferenceError: capAllLetters is not defined
+// The test failed successfully. I have a feeling that there's something wrong or out of place, I could be wrong but we will find out soon.
 
 // b) Create the function that makes the test pass.
+// Declare capAllLetters variable. Since the object being passed through the function is an array, I will need to take the array to a string using .join() so that I can then use the .toUpperCase() method to capitalize the strings. 
+    
+//This approach doesn't seem to be working. 
+
+// I think I'm going to need to run a for loop to iterate over the array to capitalize the words, and then return the new array. Since I will need to return another array, I think I'm going to need to use .map to be able to return an altered array.
+const capAllLetters = (array) => {
+    return array.map(() => {
+
+    })
+}
```

Comment:
I think you are heading in the right direction.  I do need to see a complete function (even if it is not fully working).  Your pseudocode should be your guide, try breaking the problem into small pieces to help with that next step.